plugin/ayon_openassetio_manager/ayon.py

```diff
@@ -0,0 +1,483 @@
+# Optional AYON config.
+# For ayon_core library (optional).
+CLIENT_LAUNCHER_STORAGE_DIR_KEY = "AYON_LAUNCHER_STORAGE_DIR"
+PROJECT_NAME_KEY = "AYON_PROJECT_NAME"
+PATH_NAME_KEY = "AYON_PATH_NAME"
+TASK_NAME_KEY = "AYON_TASK_NAME"
+
+env_var_settings = [
+    SERVER_URL_KEY,
+    SERVER_API_KEY,
+    SERVER_BUNDLE_NAME_KEY,
+    CLIENT_LAUNCHER_STORAGE_DIR_KEY,
+    PROJECT_NAME_KEY,
+]
+
+
+def bootstrap_ayon_core(
+        bundle_name: str, bundles_info: BundlesInfoDict) -> None:
+    """Appends the AYON core addon and dependency directories to sys.path.
+
+    E.g. by default on linux, `~/.local/share/AYON/addons/core_1.3.2` and
+    `~/.local/share/AYON/dependency_packages/ayon_2502101346_linux-rocky9.zip/dependencies`.
+
+    This allows us to make use of the utilities in ayon_core, rather than
+    reinventing them.
+
+    Args:
+        bundle_name (str): The name of the bundle to use.
+        bundles_info (BundlesInfoDict): The bundles info dictionary, as
+            obtained from the AYON server.
+
+    Raises:
+        RuntimeError: If the bundle is not found, or if the required
+            directories do not exist.
+
+    """
+    ayon_storage_dir = pathlib.Path(
+        os.environ.get(
+            CLIENT_LAUNCHER_STORAGE_DIR_KEY,
+            platformdirs.user_data_dir("AYON", "Ynput")
+        )
+    )
+
+    bundle = next(
+        (
+            bundle for bundle in bundles_info["bundles"]
+            if bundle["name"] == bundle_name
+        ),
+        None,
+    )
+    if bundle is None:
+        msg = (f"Bundle '{bundle_name}' not found. "
+               "Please check your AYON_BUNDLE_NAME setting.")
+        raise RuntimeError(msg)
+
+    ayon_addons_dir = ayon_storage_dir / "addons"
+    if not ayon_addons_dir.is_dir():
+        msg = (f"AYON addons directory not found at '{ayon_addons_dir}'. "
+               "Please ensure you have run the AYON launcher at least once.")
+        raise RuntimeError(msg)
+
+    core_addon_dir = ayon_addons_dir / f"core_{bundle['addons']['core']}"
+    if not core_addon_dir.is_dir():
+        msg = (f"Core addon directory not found at '{core_addon_dir}'. "
+               "Please ensure you have run the AYON launcher at least once.")
+        raise RuntimeError(msg)
+
+    core_addon_vendor_dir = core_addon_dir / "ayon_core" / "vendor" / "python"
+
+    ayon_dependency_packages_dir = ayon_storage_dir / "dependency_packages"
+    if not ayon_dependency_packages_dir.is_dir():
+        msg = (f"AYON dependency packages directory not found at "
+               f"'{ayon_dependency_packages_dir}'. "
+               "Please ensure you have run the AYON launcher at least once.")
+        raise RuntimeError(msg)
+
+    ayon_dependency_packages_dir /= bundle["dependencyPackages"][sys.platform]
+    ayon_dependency_packages_dir /= "dependencies"
+    if not ayon_dependency_packages_dir.is_dir():
+        msg = (f"AYON dependency packages for platform '{sys.platform}' "
```

Comment:
we don't use `sys.platform`. instead, we use `platform.system().lower()`
```suggestion
    ayon_dependency_packages_dir /= bundle["dependencyPackages"][platform.system().lower()]
    ayon_dependency_packages_dir /= "dependencies"
    if not ayon_dependency_packages_dir.is_dir():
        msg = (f"AYON dependency packages for platform '{platform.system().lower()}' "
```